lib/literal/array.rb

```diff
@@ -71,6 +78,25 @@ def <<(value)
 		self
 	end
 
+	def &(other)
+		case other
+		when ::Array
+			Literal::Array.allocate.__initialize_without_check__(
```

Comment:
We can skip type checking here because the result is a subset of elements already in the array.